pages/plate_state.vue

```diff
@@ -0,0 +1,208 @@
+import lighthouseBiosero from '@/modules/lighthouse_service_biosero'
+
+export default {
+  name: 'PlateState',
+  components: {
+    Alert,
+  },
+  data() {
+    return {
+      barcode: '',
+      lastPlateBarcode: '',
+      plate: { source: false, destination: false },
+      filter: 'source_barcode',
+      plateFields: [
+        { key: 'row', label: '', isRowHeader: true },
+        '1',
+        '2',
+        '3',
+        '4',
+        '5',
+        '6',
+        '7',
+        '8',
+        '9',
+        '10',
+        '11',
+        '12',
+      ],
+      rowHeaders: ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H'],
+    }
+  },
+  computed: {
+    plateFilterOptions() {
+      return [
+        { text: 'Source Barcode', value: 'source_barcode' },
+        { text: 'Control Barcode', value: 'control_barcode', disabled: !this.plate.destination },
+        { text: 'Control Type', value: 'control', disabled: !this.plate.destination },
+        { text: 'Source Coordinate', value: 'source_coordinate' },
+        { text: 'Destination Coordinate', value: 'destination_coordinate' },
+        { text: 'RNA ID', value: 'rna_id' },
+        { text: 'Run ID', value: 'automation_system_run_id' },
+        { text: 'Lab ID', value: 'lab_id' },
+        { text: 'LH sample UUID', value: 'lh_sample_uuid' },
+        { text: 'Date picked', value: 'date_picked' },
+        { text: 'Date created', value: 'created_at' },
+      ]
+    },
+    calculateSourceWells() {
+      const pickedWells = this.plate.samples.filter((sample) => {
+        return sample.picked
+      }).length
+      const unpickedWells = this.plate.samples.length - pickedWells
+      const emptyWells = 96 - this.plate.samples.length
+
+      return [pickedWells, unpickedWells, emptyWells]
+    },
+    calculateDestinationWells() {
+      const controlWells = this.plate.wells.filter((well) => {
+        return well.type === "control"
+      }).length
+      const pickedWells = this.plate.wells.length - controlWells
+      const emptyWells = 96 - this.plate.wells.length
+
+      return [pickedWells, controlWells, emptyWells]
+    },
+    plateItems() {
+      if (this.plate.source) {
+        return this.sourcePlateItems()
+      } else if (this.plate.destination) {
+        return this.destinationPlateItems()
+      } else {
+        return this.rowHeaders.map((header) => {
+          return { row: header }
+        })
+      }
+    },
+  },
+  methods: {
+    async findPlate() {
+      let plate = await lighthouseBiosero.getBioseroPlate(this.barcode, 'source')
+      if (plate.success) {
+        this.plate = plate
+      } else {
+        plate = await lighthouseBiosero.getBioseroPlate(this.barcode, 'destination')
+        if (plate.success) {
+          this.plate = plate
+        } else {
+          this.plate = { source: false, destination: false }
+          this.$refs.alert.show(`Could not find plate in Biosero with barcode: ${this.barcode}`, 'danger')
```

Comment:
Message sounds a bit odd. Maybe:
'Could not find a plate used on a Biosero system with the barcode: ${this.barcode}